src/context/authStore.tsx

```diff
@@ -0,0 +1,182 @@
+import React, {
+  createContext,
+  FunctionComponent,
+  useState,
+  useEffect,
+  useCallback
+} from "react";
+import { AsyncStorage } from "react-native";
+import { usePrevious } from "../hooks/usePrevious";
+import { AuthCredentials } from "../types";
+import { Sentry } from "../utils/errorTracking";
+
+export const AUTH_CREDENTIALS_STORE_KEY = "AUTH_STORE";
+
+/** @deprecated */
+export const SESSION_TOKEN_KEY = "SESSION_TOKEN";
+
+/** @deprecated */
+export const EXPIRY_KEY = "EXPIRY_KEY";
+
+/** @deprecated */
+export const ENDPOINT_KEY = "ENDPOINT_KEY";
+
+type AuthCredentialsMap = {
+  [key: string]: AuthCredentials;
+};
+
+interface AuthStoreContext {
+  hasLoadedFromStore: boolean;
+  authCredentials: AuthCredentialsMap;
+  addAuthCredentials: (key: string, authCredentials: AuthCredentials) => void;
+  removeAuthCredentials: (key: string) => void;
+  clearAuthCredentials: () => void;
+}
+
+export const AuthStoreContext = createContext<AuthStoreContext>({
+  hasLoadedFromStore: false,
+  authCredentials: {},
+  addAuthCredentials: () => undefined,
+  removeAuthCredentials: () => undefined,
+  clearAuthCredentials: () => undefined
+});
+
+export const AuthStoreContextProvider: FunctionComponent<{
+  shouldMigrate?: boolean; // Temporary toggle to disable migration when testing
```

Comment:
so we'll set this to true in stg and prod? :O haha

Reply:
It's always true, just that for testing it's set to false.